test/integration/shim_int_test.go

```diff
@@ -0,0 +1,121 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/stretchr/testify/suite"
+)
+
+type ShimIntegrationTestSuite struct {
+	suite.Suite
+}
+
+func (suite *ShimIntegrationTestSuite) createProjectFile(ts *e2e.Session) {
+	ts.PrepareActiveStateYAML(strings.TrimSpace(`
+		project: https://platform.activestate.com/ActiveState-CLI/Python3?commitID=fbc613d6-b0b1-4f84-b26e-4aa5869c4e54
+	`))
+}
+
+func (suite *ShimIntegrationTestSuite) TestShim_Basic() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	suite.createProjectFile(ts)
+
+	scriptBlock := `
+print("Hello World!")
+	`
+
+	testScript := filepath.Join(fmt.Sprintf("%s/%s.pl", ts.Dirs.Work, suite.T().Name()))
+	fail := fileutils.WriteFile(testScript, []byte(scriptBlock))
+	suite.Require().NoError(fail.ToError())
+
+	cp := ts.Spawn("shim", "--", "python3", fmt.Sprintf("%s", testScript))
+	cp.Expect("Hello World!")
+	cp.ExpectExitCode(0)
```

Comment:
If this is working there's something wrong with the exit stuff. This is effectively running:

```shim -- python3 "print("Hello World!")"```

Note the double quotes.

Reply:
I'm not sure I'm following. `testScript` is the path of the file so this should be running 

`state shim -- python3 /path/to/script.py`

And the subsequent shell script that the subshell runs should be 

`python3 /path/to/script.py`

The contents of the script file should not be read and passed as a string.